compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -1336,33 +1390,27 @@ void generateArrayStoreIns(BIRNonTerminator.FieldAccess inst) {
         this.loadVar(inst.keyOp.variableDcl);
         this.loadVar(inst.rhsOp.variableDcl);
 
-        BType valueType = inst.rhsOp.variableDcl.type;
-        BType varRefType = inst.lhsOp.variableDcl.type;
-        if (varRefType.tag == TypeTags.JSON ||
-                (varRefType.tag == TypeTags.ARRAY && ((BArrayType) varRefType).eType instanceof BJSONType)) {
-            addBoxInsn(this.mv, valueType);
-            this.mv.visitMethodInsn(INVOKESTATIC, JSON_UTILS, "setArrayElement",
-                    String.format("(L%s;JL%s;)V", OBJECT, OBJECT), false);
-            return;
-        }
-
-        if (TypeTags.isIntegerTypeTag(valueType.tag)) {
-            this.mv.visitMethodInsn(INVOKEINTERFACE, ARRAY_VALUE, "add", "(JJ)V", true);
-        } else if (valueType.tag == TypeTags.FLOAT) {
-            this.mv.visitMethodInsn(INVOKEINTERFACE, ARRAY_VALUE, "add", "(JD)V", true);
-        } else if (TypeTags.isStringTypeTag(valueType.tag)) {
-            this.mv.visitMethodInsn(INVOKEINTERFACE, ARRAY_VALUE, "add",
-                    String.format("(JL%s;)V", JvmPackageGen.IS_BSTRING ? JvmConstants.B_STRING_VALUE :
-                            STRING_VALUE), true);
-        } else if (valueType.tag == TypeTags.BOOLEAN) {
-            this.mv.visitMethodInsn(INVOKEINTERFACE, ARRAY_VALUE, "add", "(JZ)V", true);
-        } else if (valueType.tag == TypeTags.BYTE) {
-            this.mv.visitInsn(I2B);
-            this.mv.visitMethodInsn(INVOKEINTERFACE, ARRAY_VALUE, "add", "(JB)V", true);
-        } else {
-            this.mv.visitMethodInsn(INVOKEINTERFACE, ARRAY_VALUE, "add", String.format("(JL%s;)V", OBJECT), true);
+            BType valueType = inst.rhsOp.variableDcl.type;
+
+            String method = "add";
+
+            if (TypeTags.isIntegerTypeTag(valueType.tag)) {
+                this.mv.visitMethodInsn(INVOKEINTERFACE, ARRAY_VALUE, method, "(JJ)V", true);
+            } else if (valueType.tag == TypeTags.FLOAT) {
+                this.mv.visitMethodInsn(INVOKEINTERFACE, ARRAY_VALUE, method, "(JD)V", true);
+            } else if (TypeTags.isStringTypeTag(valueType.tag)) {
+                this.mv.visitMethodInsn(INVOKEINTERFACE, ARRAY_VALUE, method,
+                                        String.format("(JL%s;)V", JvmPackageGen.IS_BSTRING ?
+                                                JvmConstants.B_STRING_VALUE : STRING_VALUE), true);
+            } else if (valueType.tag == TypeTags.BOOLEAN) {
+                this.mv.visitMethodInsn(INVOKEINTERFACE, ARRAY_VALUE, method, "(JZ)V", true);
+            } else if (valueType.tag == TypeTags.BYTE) {
+                this.mv.visitInsn(I2B);
+                this.mv.visitMethodInsn(INVOKEINTERFACE, ARRAY_VALUE, method, "(JB)V", true);
+            } else {
+                this.mv.visitMethodInsn(INVOKEINTERFACE, ARRAY_VALUE, method, String.format("(JL%s;)V", OBJECT), true);
+            }
```

Comment:
looks like there is an alignment issue here. probably due to merge conflicts with jvm codegen package classes.